internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,76 @@
 package daysteps
 
+import (
+	"fmt"
+
+	"github.com/Yandex-Practicum/tracker/internal/personaldata"
+
+	// "internal/personaldata"
+	"github.com/Yandex-Practicum/tracker/internal/spentenergy"
+	//"internal/spentenergy"
+	"strconv"
+	"strings"
+	"time"
+)
+
 type DaySteps struct {
-	// TODO: добавить поля
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
 }
 
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+	// 1. разделить строку на слайс строк
+	slice := strings.Split(datastring, ",")
+
+	// 2. проверить, чтобы длина слайса была равна 2, так как в строке данных у нас количество шагов и продолжительность
+	if len(slice) != 2 {
+		return fmt.Errorf("длина слайса не равна 2")
+	}
+
+	// 3. преобразовать первый элемент слайса (количество шагов) в тип int. Обработать возможные ошибки. При их возникновении
+	// из функции вернуть 0 шагов, 0 продолжительность и ошибку
+	steps, err := strconv.Atoi(slice[0])
+	if err != nil {
+		return fmt.Errorf("не удалось преобразовать количество шагов: %v", err)
+	}
+
+	// 4. проверить: количество шагов должно быть больше 0. Если это не так, вернуть нули и ошибку
+	if steps <= 0 {
+		return fmt.Errorf("количество шагов должно быть больше 0")
+	}
+
+	// 5. преобразовать второй элемент слайса в time.Duration. В пакете time есть метод для парсинга строки в time.Duration.
+	// Обработать возможные ошибки. При их возникновении из функции вернуть 0 шагов, 0 продолжительность и ошибку
+	duration, err := time.ParseDuration(slice[1])
+	if err != nil {
+		return fmt.Errorf("не удалось преобразовать продолжительность: %v", err)
+	}
+	if duration <= 0 {
+		return fmt.Errorf("продолжительность должна быть больше 0")
+	}
+
+	// 6. если всё прошло без ошибок, записать количество шагов и продолжительность, вернуть nil (для ошибки)
+	ds.Steps = steps
+	ds.Duration = duration
+	return nil
 }
 
 func (ds DaySteps) ActionInfo() (string, error) {
-	// TODO: реализовать функцию
+	// 1. Вычислите дистанцию
+	if ds.Steps <= 0 {
+		return "", fmt.Errorf("количество шагов должно быть больше нуля")
+	}
+	dist := spentenergy.Distance(ds.Steps, ds.Personal.Height)
+	// 2. Вычислите количество сожжённых калорий. При возникновении ошибки верните пустую строку и ошибку
+	calories, err := spentenergy.WalkingSpentCalories(ds.Steps, ds.Personal.Weight, ds.Personal.Height, ds.Duration)
+	if err != nil {
+		return "", fmt.Errorf("%v", err)
```

Comment:
Лучше всегда оборачивать
```suggestion
		return "", fmt.Errorf("error calculating calories: %w", err)
```